comp/core/workloadmeta/collectors/internal/remote/processcollector/process_collector_test.go

```diff
@@ -232,8 +236,24 @@ func TestCollection(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			mockConfig := config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
-			mockConfig.Set("language_detection.enabled", true)
+
+			overrides := map[string]interface{}{
+				"language_detection.enabled": true,
+			}
+
+			// We do not inject any collectors here; we instantiate
+			// and initialize it out-of-band below. That's OK.
+			mockStore := fxutil.Test[workloadmeta.Mock](t, fx.Options(
+				core.MockBundle,
+				fx.Replace(config.MockParams{Overrides: overrides}),
+				fx.Supply(workloadmeta.Params{
+					AgentType: workloadmeta.Remote,
+				}),
+				workloadmeta.MockModuleV2,
+			))
+
+			time.Sleep(time.Second)
```

Comment:
I want to remove some of the sleeps all over the place, I agree that this is only going to cause likely flakes in the future. But this is indeed probably the easiest for now, doing it otherwise presents some challenges I think.